src/snowflake/snowpark/_internal/compiler/query_generator.py

```diff
@@ -169,6 +182,7 @@ def do_resolve_with_resolved_children(
                 iceberg_config=logical_plan.iceberg_config,
                 table_exists=logical_plan.table_exists,
             )
+            resolved_plan.referenced_ctes = resolved_child.referenced_ctes
```

Comment:
i do not expect we need to do the propogation here if the plan is all resovled correctly